doc/source/user/basics.copies.rst

```diff
@@ -0,0 +1,144 @@
+.. _basics.copies-and-views:
+
+****************
+Copies and views
+****************
+
+When operating on a NumPy array, we can sometimes improve performance by using
+:term:`views <view>` instead of copies of the array. However, when handling
+large amounts of important data, it can be risky to directly operate on it and
+it could be safer to use copies. Hence, it is important to know the difference
+between these two terms to correctly determine when to use which one.
+
+The NumPy array is a :term:`contiguous` block of memory consisting of two parts:
+the data buffer with the actual data elements, and the metadata which contains
+information about the data buffer. The metadata includes data type, strides
+and other important information that helps manipulate the :class:`.ndarray`
+easily. See the :ref:`numpy-internals` section for a detailed look.
+
+View or shallow copy
+====================
+
+It is possible to access the array differently by just changing the
+metadata and without changing the data buffer. This creates a new way of
+looking at the data and these new arrays are called views or shallow copies.
+The data buffer remains the same so any changes made to a view reflects in the
+original copy. A view can be forced through the :meth:`.ndarray.view` method. 
+
+
+Copy or deep copy
+=================
+
+When a new array is created by duplicating the data buffer as well as the
+metadata, it is called a copy or a deep copy. Changes made to the copy
+do not reflect on the original array. Making a copy is slower and memory
+consuming but sometimes necessary. A copy can be forced by using
+:meth:`.ndarray.copy`.
```

Comment:
I think we should be careful with the shallow/deep copy distinction here - I think it's actually confusing to conflate views/copies (in the numpy context) with shallow/deep copying (in the broader Python sense). For example - `ndarray.copy` itself is actually a *shallow* copy (relevant for object arrays containing mutable objects):

```python
>>> a = np.array([1, 2, [3, 4, 5]], dtype=object)
>>> b = a.copy()
>>> b[2][1] = 10
>>> a
array([1, 2, list([3, 10, 5])], dtype=object)
```

IMO the easiest way to reduce the confusion here is to avoid the analogy to shallow/deep copies.